src/tribol/interface/tribol.cpp

```diff
@@ -921,6 +1110,96 @@ integer update( integer cycle, real t, real &dt )
       }
 
       CouplingScheme* couplingScheme  = csManager.getCoupling(csIndex);
+
+      // update redecomp meshes if supplied mfem data
+      if (couplingScheme->hasMfemData())
+      {
+         auto mfem_data = couplingScheme->getMfemMeshData();
+         axom::Array<int> mesh_ids {2, 2};
+         mesh_ids[0] = mfem_data->GetMesh1ID();
+         mesh_ids[1] = mfem_data->GetMesh2ID();
+         mfem_data->UpdateMeshData();
+         auto coord_ptrs = mfem_data->GetRedecompCoordsPtrs();
+
+         // save penalty data for old meshes
+         auto& mesh_manager = MeshManager::getInstance();
+         axom::Array<MeshElemData> mesh_data {0, 2};
+         mesh_data.push_back(mesh_manager.GetMeshInstance(mesh_ids[0]).m_elemData);
+         mesh_data.push_back(mesh_manager.GetMeshInstance(mesh_ids[1]).m_elemData);
+
+         registerMesh(
+            mesh_ids[0],
+            mfem_data->GetMesh1NE(),
+            mfem_data->GetNV(),
+            mfem_data->GetMesh1Conn(),
+            mfem_data->GetElemType(),
+            coord_ptrs[0],
+            coord_ptrs[1],
+            coord_ptrs[2]
+         );
+         registerMesh(
+            mesh_ids[1],
+            mfem_data->GetMesh2NE(),
+            mfem_data->GetNV(),
+            mfem_data->GetMesh2Conn(),
+            mfem_data->GetElemType(),
+            coord_ptrs[0],
+            coord_ptrs[1],
+            coord_ptrs[2]
+         );
+
+         // apply penalty data
+         for (int i {0}; i < 2; ++i)
+         {
+            auto& new_mesh_data = mesh_manager.GetMeshInstance(mesh_ids[i]).m_elemData;
+            new_mesh_data.m_is_kinematic_constant_penalty_set
+               = mesh_data[i].m_is_kinematic_constant_penalty_set;
+            new_mesh_data.m_is_kinematic_element_penalty_set
+               = mesh_data[i].m_is_kinematic_element_penalty_set;
+            new_mesh_data.m_is_rate_constant_penalty_set
+               = mesh_data[i].m_is_rate_constant_penalty_set;
+            new_mesh_data.m_is_rate_percent_penalty_set
+               = mesh_data[i].m_is_rate_percent_penalty_set;
+            new_mesh_data.m_penalty_scale = mesh_data[i].m_penalty_scale;
+            new_mesh_data.m_penalty_stiffness = mesh_data[i].m_penalty_stiffness;
+            new_mesh_data.m_rate_penalty_stiffness
+               = mesh_data[i].m_rate_penalty_stiffness;
+            new_mesh_data.m_rate_percent_stiffness
+               = mesh_data[i].m_rate_percent_stiffness;
+            SLIC_ERROR_ROOT_IF(mesh_data[i].m_is_kinematic_element_penalty_set,
+               "Element penalties not supported for MFEM meshes.");
```

Comment:
Let's add task for next PR to support the element-wise penalty.

Reply:
Added an issue for this.